contexts/AutocratContext.tsx

```diff
@@ -40,93 +34,193 @@ export const useAutocrat = () => {
-    queryKey: ['getDao'],
-    queryFn: () => autocratProgram.account.dao.fetch(dao),
+  // TODO: We need to surface an option to select a DAO, so there's
+  // some point of reference from the user to allow us to sort on this.
+  // I've setup a list of DAOs hard coded, but this is still missing a
+  // selection given you may have many starting at v0.3
+  const daosQueryKey = `${network}-${programVersion.label}-${programVersion.dao}-daolist`;
+  const { data: daos } = useQuery({
+    queryKey: [daosQueryKey],
+    queryFn: () => autocratProgram.account.dao.all(),
     staleTime: 30_000,
     refetchOnMount: false,
   });
-  const daoState = daoStateData;
+
+  // TODO: THIS NEEDS TO BE HANDLED WHEN MULTI DAO
+  // Filter against our list of DAOs
+  let selectedDao = daos?.find(
+    (dao) => dao.publicKey.toString() === DAOS[0].publicKey.toString()
+  );
+
+  // Need to use any DAO even if we didn't get a match from our list
+  if (!selectedDao && daos) {
+    [selectedDao] = daos;
+  }
+
+  const daoKey = selectedDao?.publicKey;
+
+  const daoTreasuryKey = selectedDao?.account.treasury;
+
+  const daoState = selectedDao?.account;
+
   const [proposals, setProposals] = useState<Proposal[]>();
 
-  const fetchProposals = useCallback(async () => {
-    const props = ((await autocratProgram?.account.proposal?.all()) || []).sort((a, b) =>
+  const proposalsQueryKey = `${network}-${programVersion.label}-${programVersion.dao}-proposals`;
+  const { data: allProposals } = useQuery({
+    queryKey: [proposalsQueryKey],
+    queryFn: () => autocratProgram?.account.proposal?.all(),
+    staleTime: 30_000,
+    refetchOnMount: false,
+  });
+
+  // Moved token stuff into the Autocrat, given this is where we reference core tokens
+  const defaultTokens: TokensDict = useMemo(() => {
+    switch (network) {
+      case Networks.Devnet:
+        return { ...staticTokens, ...devnetTokens };
+      case Networks.Mainnet:
+        return { ...staticTokens, ...mainnetTokens };
+      case Networks.Custom:
+        return { ...staticTokens, ...mainnetTokens, ...devnetTokens };
+      default:
+        // TODO: Dunno, got stuck on this.
+        return { ...staticTokens, ...mainnetTokens, ...devnetTokens };
+    }
+  }, [network, programVersion.label]);
+
+  const [daoTokens, setDaoTokens] = useLocalStorage<TokensDict>({
+    key: 'futarchy-tokens',
+    defaultValue: defaultTokens,
+    getInitialValueInEffect: true,
+    serialize: JSON.stringify,
+    deserialize: (s) => {
+      if (!s) return {};
+      const o: TokensDict = JSON.parse(s);
+      return Object.fromEntries(
+        Object.entries(o).map(([k, v]: [string, Token]) => [
+          k,
+          { ...v, publicKey: new PublicKey(v.publicKey) },
+        ]),
+      );
+    },
+  });
+
+  // TODO: The goal here is the fetch the token data to enrich it.
+  // all we need is a URI for display within the application
+  // NOTE: This is not working.
+  // const tokenMetadata = useQueries({
```

Comment:
do we wanna keep this around though?

Reply:
I just wanted an example / reference, in the old stuff as it could be nice so we don't have to manage as much, but also as a ? for why it wasn't working.